src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Caching.cs

```diff
@@ -86,8 +84,7 @@ internal void ClearCaches()
         [MemberNotNull(nameof(_cachingContext))]
         private void InitializeCachingContext()
         {
-            _isLockedInstance = true;
-            _cachingContext = TrackedCachingContexts.GetOrCreate(this);
+            _cachingContext ??= TrackedCachingContexts.GetOrCreate(this);
```

Comment:
no need to lock here? if it's already locked should we at least add assert? We should not be doing this if it's not locked